### PR TITLE
Add integration to travis-ci

### PR DESCRIPTION
**Pull request by woodpigeon**
_This is the re-creation of  #21 to simplify merging_

We believe its important to build CI into your development process, and use [Travis-CI](https://travis-ci.org/) for our open source projects as its free, flexible and simple to use.

This change adds everything required to integrate the project with Travis-CI, and get it automatically building our PR's and branches. It includes

- addition of a `.travis.yml` build config file
- adding a badge to `README.md` to display the current build status of the project

### CODE REVIEW

**Comment by woodpigeon** [comment]
The latest status of this is that @reggieb has changed before_commit to add a `run_without_checks` to get around the before_commit:run failure 'not a git repo'. However it is taking a while for travis to pick up the new gem from rubgems so this on hold until it can download before_commit 0.8.0.

**Comment by woodpigeon** [comment]
Now picking up latest before_commit at ready to 👀

File: spec/dummy/config/database.travis.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# In travis-ci this file is copied over the database.yml before the tests are run.
+# As per https://docs.travis-ci.com/user/database-setup/#SQLite3 if you use activerecord they recommend adding the
+# following configuration for your test database.
+test:
```
Comment (Cruikshanks) [inline spec/dummy/config/database.travis.yml]: Just checking this file is still needed based on the current details in `.travis.yml`. It seems to be setting up the CI environment for using Postgres and we are not calling the command to overwrite `database.yml` with this. So would it b clearer to just delete this file now?

**Comment by Cruikshanks** [comment]
Please also don't take my above comment as a full PR of the code. Its just a thought having quickly scanned the latest changes following my original creation of the PR.

File: flood_risk_engine.gemspec [hunk 2]
```diff
@@ -22,11 +22,11 @@ Gem::Specification.new do |s|
   s.add_dependency "reform-rails", "~> 0.1"
   s.add_dependency "dotenv-rails", "~> 2.1"
   s.add_dependency "finite_machine", "~> 0.10"
-  s.add_dependency "dibber", "~> 0.5" # Manages data seeding
+  s.add_dependency "dibber", "~> 0.4" # Manages data seeding
```
Comment (reggieb) [inline flood_risk_engine.gemspec]: You'll need dibber 0.5 for the syntax in `db:/seeds.rb` to work.

**Comment by reggieb** [comment]
Looks fine except the dibber version. I think if you try `rake db:seed` at `spec/dummy` it will fail with that configuration.

Otherwise :ship: it

**Pull request merged**